File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/storage/data/entities/enemies/units/ShaKireFirstSwordsman.java
package it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.units;

import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.Skills;
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy;
import it.paranoidsquirrels.idleguildmaster.storage.data.items.ItemWrapper;
import java.util.LinkedHashMap;

/* JADX INFO: loaded from: classes3.dex */
public class ShaKireFirstSwordsman extends Enemy {
    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected int getMaxDamage() {
        return 104;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected int getMinDamage() {
        return 75;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.Entity
    public boolean isMagic() {
        return false;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.Entity
    public boolean isRanged() {
        return false;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected void configureStatistics() {
        this.baseMaxHp = 3500;
        this.baseConstitution = 24;
        this.baseIntelligence = 36;
        this.baseDexterity = 42;
        this.baseDefense = 0;
        this.baseMagicDefense = 0;
        this.counterattack = 1.0d;
        this.imageId = R.drawable.unit_sha_kire_first_swordsman;
        this.idName = R.string.enemy_sha_kire_first_swordsman_name;
        this.idDescription = R.string.enemy_sha_kire_first_swordsman_description;
        this.passiveSkill = Skills.PASSIVE_RETALIATE_II;
        this.activeSkill = Skills.ACTIVE_DESERT_JUDGEMENT;
        this.expGiven = 600;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    public LinkedHashMap<ItemWrapper, Integer> listDrops(int i) {
        LinkedHashMap<ItemWrapper, Integer> linkedHashMap = new LinkedHashMap<>();
        linkedHashMap.put(ItemWrapper.getInstance("EyesOfTheSwordsman", 1), 1000);
        return linkedHashMap;
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/storage/data/entities/enemies/units/ShaKireFirstSwordsman.kt
package it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.units

import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.Skills
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
import it.paranoidsquirrels.idleguildmaster.storage.data.items.ItemWrapper
import java.util.LinkedHashMap

class ShaKireFirstSwordsman : Enemy() {
    override fun getMaxDamage(): Int = 104
    override fun getMinDamage(): Int = 75
    override fun isMagic(): Boolean = false
    override fun isRanged(): Boolean = false

    override fun configureStatistics() {
        baseMaxHp = 3500
        baseConstitution = 24
        baseIntelligence = 36
        baseDexterity = 42
        baseDefense = 0
        baseMagicDefense = 0
        counterattack = 1.0
        imageId = R.drawable.unit_sha_kire_first_swordsman
        idName = R.string.enemy_sha_kire_first_swordsman_name
        idDescription = R.string.enemy_sha_kire_first_swordsman_description
        passiveSkill = Skills.PASSIVE_RETALIATE_II
        activeSkill = Skills.ACTIVE_DESERT_JUDGEMENT
        expGiven = 600
    }

    override fun listDrops(i: Int): LinkedHashMap<ItemWrapper, Int> {
        val linkedHashMap = LinkedHashMap<ItemWrapper, Int>()
        linkedHashMap.put(ItemWrapper.getInstance("EyesOfTheSwordsman", 1), 1000)
        return linkedHashMap
    }
}
